### ISSUE TITLE
Adapter gets SIGSEGV or SIGABRT and is restarted .....

### ISSUE THREAD

**Issue by jhergeth** [frankjoke/ioBroker.radar#29]
Hello,

a newly installed radar adapter generates:
`
host.iob | 2019-01-26 00:19:35.365 | info | Restart adapter system.adapter.radar.0 because enabled

host.iob | 2019-01-26 00:19:35.365 | error | instance system.adapter.radar.0 terminated with code null ()

host.iob | 2019-01-26 00:19:35.365 | warn | instance system.adapter.radar.0 terminated due to SIGSEGV

radar.0 | 2019-01-26 00:19:34.707 | debug | Make State ExternalNetwork.status and set value to 2

radar.0 | 2019-01-26 00:19:34.699 | debug | Make State ExternalNetwork.IP4 and set value to '80.141.178.231'

radar.0 | 2019-01-26 00:19:34.699 | debug | Same IP address 80.141.178.231

radar.0 | 2019-01-26 00:19:34.035 | info | Init item Fritzbox with { name: 'Fritzbox', ip: 'fritz.box', macs: '', bluetooth: '', id: 'Fritzbox', hasBT: false, printer: false, hasIP: true }

radar.0 | 2019-01-26 00:19:33.984 | info | Init item Ich- with { name: 'Ich-', ip: 'localhost', macs: '', bluetooth: '', id: 'Ich', hasBT: false, printer: false, hasIP: true }

radar.0 | 2019-01-26 00:19:33.945 | info | BT Bin Dir = '/opt/iobroker/node_modules/iobroker.radar\bin\bluetoothview\'

radar.0 | 2019-01-26 00:19:33.945 | info | radar set to scan every 60 sec and printers every 100 scans.

radar.0 | 2019-01-26 00:19:33.945 | info | found '@abandonware/noble'

radar.0 | 2019-01-26 00:19:33.943 | info | starting. Version 1.2.1 in /opt/iobroker/node_modules/iobroker.radar, node: v8.15.0

host.iob | 2019-01-26 00:19:33.183 | info | instance system.adapter.radar.0 started with pid 3731

host.iob | 2019-01-26 00:19:03.173 | info | Restart adapter system.adapter.radar.0 because enabled

host.iob | 2019-01-26 00:19:03.173 | error | instance system.adapter.radar.0 terminated with code null ()

host.iob | 2019-01-26 00:19:03.172 | warn | instance system.adapter.radar.0 terminated due to SIGSEGV

radar.0 | 2019-01-26 00:19:02.475 | debug | Make State ExternalNetwork.status and set value to 2

radar.0 | 2019-01-26 00:19:02.466 | debug | Make State ExternalNetwork.IP4 and set value to '80.141.178.231'

radar.0 | 2019-01-26 00:19:02.466 | debug | Same IP address 80.141.178.231

radar.0 | 2019-01-26 00:19:01.981 | info | Init item Fritzbox with { name: 'Fritzbox', ip: 'fritz.box', macs: '', bluetooth: '', id: 'Fritzbox', hasBT: false, printer: false, hasIP: true }

radar.0 | 2019-01-26 00:19:01.930 | info | Init item Ich- with { name: 'Ich-', ip: 'localhost', macs: '', bluetooth: '', id: 'Ich', hasBT: false, printer: false, hasIP: true }

radar.0 | 2019-01-26 00:19:01.898 | info | BT Bin Dir = '/opt/iobroker/node_modules/iobroker.radar\bin\bluetoothview\'

radar.0 | 2019-01-26 00:19:01.898 | info | radar set to scan every 60 sec and printers every 100 scans.

radar.0 | 2019-01-26 00:19:01.898 | info | found '@abandonware/noble'

radar.0 | 2019-01-26 00:19:01.897 | info | starting. Version 1.2.1 in /opt/iobroker/node_modules/iobroker.radar, node: v8.15.0

host.iob | 2019-01-26 00:19:01.134 | info | instance system.adapter.radar.0 started with pid 3706
`

Adapter is the current version 1.2.1.
IOBroker is current and runs in a docker environment on a synology.
I tried several reinstalls without any luck.
I disabled dwz by setting the timer to 0, adapter is still resetting.

Any suggestions?
JHeg

**Comment by frankjoke** [comment]
Ok, I am not so used with docker images but the error line shows that a certain C program is generating the error.
This might be arp-scan, ping or fping but I cannot see which, it just shows that it's something called from node but not node itself.

p.s.: I just write on an complete adapter rewrite to remove need for fping or ping but support of IpV6 , just arp-scan will still be used if available. This will reduce traffic and cpu-load.

p.s.: my Development machine is a virtual (x86) Stretch image running a QNAP NAS - and I test then on Raspi, Orangepi and NUK before I post it in here!

**Comment by jhergeth** [comment]
Hi,
I also thought about a rewrite eliminating the dependency on external programs. I found https://www.npmjs.com/package/net-ping which seems to be good replacement for the IP-Ping part. I would also try to split the tool into different parts (radarIP, radarBT and radarWeather). Users could then choose which they would use.

I am in no way familiar with the iobroker coding, I looked into your sources and did not understand the code. So I immediately gave up ;-).

If you have any testable code I can surely test it in my Synology/Docker environment.

**Comment by frankjoke** [comment]
jhergeth,
that's what I actually am testing now, the net-ping is working on all my Linux so far but I could not test on windows. I also changed some other tools I used, for example the website to get the MAC-Vendor. I found another tool which I want to integrate as well: It looks at DHCP packages and could make somebody active as soon his device logs into wlan, not only after a scan! So it's a complete re-code more or less because some of the new things are not possible with current structure of scans. I need to update also the admin page dramatically.
